### PR TITLE
[FEA] Remove FAISS dependency, inherit other common dependencies from raft

### PR DESCRIPTION
**Pull request by trxcllnt**
Originally this PR was about building FAISS shared libs via CPM, but @rlratzel mentioned cuGraph doesn't need FAISS anymore, and it'd be better if we remove it.

If we need FAISS again in the future, we can add `faiss::faiss` back to `target_link_libraries` without needing extra CPM configuration as it will be available [via raft](https://github.com/rapidsai/raft/pull/345).

Edit: Removed more dependencies that we inherit from raft and/or rmm. Depends on https://github.com/rapidsai/raft/pull/345.

Edit 2: I think the CUDA 11.0 thrust issue will be solved by https://github.com/rapidsai/rapids-cmake/pull/98

### CODE REVIEW

**Comment by BradReesWork** [comment]
@trxcllnt is this really a hotfix for 21.10?

**Comment by ajschmidt8** [comment]
rerun tests

File: cpp/cmake/thirdparty/get_faiss.cmake
Comment (robertmaynard) [inline cpp/cmake/thirdparty/get_faiss.cmake:40]: No need the variable will get propagated automatically

File: cpp/cmake/thirdparty/get_faiss.cmake
Comment (robertmaynard) [inline cpp/cmake/thirdparty/get_faiss.cmake:50]: Since FAISS is a private dependency of cugraph why do we need to install / export this at all?

**Comment by BradReesWork** [comment]
@trxcllnt by pushing dependencies to RAFT, how will cuGraph know what it is dependent on and what libraries are being brought into the final package?

**Comment by trxcllnt** [comment]
rerun tests

**Comment by trxcllnt** [comment]
rerun tests

**Comment by trxcllnt** [comment]
rerun tests

**Comment by trxcllnt** [comment]
rerun tests

File: build.sh
Comment (rlratzel) [inline build.sh:27]: Can we remove this commented-out block, or is this something that might still be needed?

File: cpp/CMakeLists.txt
Comment (rlratzel) [inline cpp/CMakeLists.txt:435]: is it better to do this, or just remove it altogether along with the `FINAL_CODE_BLOCK` args to the `rapids_export` calls below?

File: cpp/cmake/thirdparty/get_raft.cmake
Comment (rlratzel) [inline cpp/cmake/thirdparty/get_raft.cmake:48]: I'm assuming this is just for debugging the CI failures?

**Review by rlratzel — COMMENTED** [review]
Thank you for also remembering to update `build.sh`.

File: cpp/cmake/thirdparty/get_raft.cmake
Comment (trxcllnt) [reply cpp/cmake/thirdparty/get_raft.cmake:48]: Without this, CI in this PR would never go green :-)

File: build.sh
Comment (trxcllnt) [reply build.sh:27]: yes this isn't needed anymore. I'll remove it.

File: cpp/CMakeLists.txt
Comment (trxcllnt) [reply cpp/CMakeLists.txt]: Removed!

File: cpp/CMakeLists.txt
Comment (cjnolet) [reply cpp/CMakeLists.txt:149]: @ChuckHastings, isnt cugraph is still using the RAFT comms?

Comment (ChuckHastings) [reply cpp/CMakeLists.txt:149]: Yes, cugraph uses the RAFT comms implementation.

Comment (cjnolet) [reply cpp/CMakeLists.txt:149]: Ah, I misunderstood the original statement- you were saying cugraph should be getting nccl transitively through raft.

**Review by ajschmidt8 — APPROVED** [review]
Approving `ops-codeowner` file changes

**Comment by rlratzel** [comment]
rerun tests

Rerunning to get the latest updates to fix style checker version check and CMakeLists.txt changes that resolve a build failure.

File: ci/gpu/build.sh
Comment (rlratzel) [inline ci/gpu/build.sh:94]: oops.

**Comment by trxcllnt** [comment]
@robertmaynard and I discussed offline that we need to add a `FAISS` component to raft before we can merge this, as we don't want CMake to download and build faiss when cuGraph doesn't use it.

**Comment by trxcllnt** [comment]
rerun tests

**Comment by trxcllnt** [comment]
rerun tests

**Comment by trxcllnt** [comment]
rerun tests

**Comment by trxcllnt** [comment]
rerun tests

**Comment by trxcllnt** [comment]
rerun tests

**Comment by trxcllnt** [comment]
rerun tests

**Comment by trxcllnt** [comment]
rerun tests

**Comment by cjnolet** [comment]
We recently merged the PR to raft that establishes multiple output targets- specifically, there is now a libraft-nn package which depends kn FAISS (and which cugraph no longer depends on directly). Does this help at all alleviate the FAISS issue on cugraph?

**Comment by trxcllnt** [comment]
rerun tests

**Comment by cjnolet** [comment]
@trxcllnt, not sure if it helps but I opened https://github.com/rapidsai/cugraph/pull/2009 to fully remove the faiss dependency from cugraph and implement @robertmaynard's recent changes to the raft build.

**Comment by trxcllnt** [comment]
rerun tests

**Comment by trxcllnt** [comment]
rerun tests

**Comment by trxcllnt** [comment]
This PR has been updated to use https://github.com/rapidsai/raft/pull/428

**Comment by trxcllnt** [comment]
rerun tests

**Comment by cjnolet** [comment]
rerun tests

**Comment by cjnolet** [comment]
rerun tests

**Comment by trxcllnt** [comment]
rerun tests

**Comment by BradReesWork** [comment]
@gpucibot merge

**Pull request merged**
